### ISSUE TITLE
Add button to/from script

### ISSUE THREAD

**Issue by ggrrll** [ipython-contrib/jupyter_contrib_nbextensions#869]
As sometimes it's just faster to work with a .py script, would be nice to quickly being able to generate it  from the notebook where you are working on: add **'to script' button**.   
On the other side, we want to make readable project that have been developed only in scripts, so also a  **'new notebook from scipt' button** would be appreciated.

Thanks for your efforts.

**Comment by jfbercher** [comment]
For saving to `.py` format, you have `File|Download as|Python`. For loading a Python file, you can use the `%%load` magic. Do I miss something?

**Comment by ggrrll** [comment]
yeah, I know - sorry,  I should have been more clear.
I thought about something that recognise comments/large space between code and comments, 
and split automatically the cell, adding the next comment a md-cell, and the following  code in a code-cell

- if you load a long script, to make it readable for other (students, etc...) you have to do that manually

ps: it's something a bit fancy...but it might be useful in some cases

**Comment by jcb91** [comment]
@ggrrll so I guess you mean something like

```python
# An example python script which can be converted to a notebook.
# Comments are fine!

# ============================== cell-boundary ============================== #

message = 'hello this is a variable in the first cell'

# ============================== cell-boundary ============================== #

print(message)

```

in script, which can be split at the comments?

**Comment by ggrrll** [comment]
yeah... - I would prefer a quite flexible detection of the cell boundary (~ any comments)

ps: what I have in mind (like many others I guess) is to use the notebook to reproduce the computations done in research papers. 
This addition would make life easier when starting a 'simple'  script (at least for those languages for which a notebook kernel exists), that has not been   really written for 'didactic purposes'.
(now, more and more researchers attach  `.py` _and_ `.ipynb`, but still...)

**Comment by jcb91** [comment]
> yeah... - I would prefer a quite flexible detection of the cell boundary (~ any comment)

ok, so then any comment would become a markdown cell? Or just stay as a comment with the code following it? I guess this behaviour could be made configurable anyway.

My other question is what you envisage the workflow to be. For notebook-to-script, it makes sense to run from the notebook view (since the notebook already exists), but for the other way around (script to notebook), do you run this from the notebook view, and it replaces the current notebook contents witnew content parsed from a `.py` file of the same name? Or perhaps it makes more sense to have an option in the file-browser (tree) view, e.g. a button like the duplicate/rename ones which appear for selected files. Something like `to new notebook`?

The final thing is that because of added ipython syntax, converting from .ipynb to .py can apply some changes which aren't easy to undo, e.g. a whole bunch of conversions like `!echo system call` into something like `get_ipython().system('echo system call')`, in [IPython/core/inputtransformer.py](https://github.com/ipython/ipython/blob/master/IPython/core/inputtransformer.py). I haven't found obvious code for the reverse transformations, and indeed, it's presumably not really possible to tell whether somebody actually meant to write the python in their .py script, rather than the ipython equivalent. Anyway, the upshot is that roundtrip conversion will introduce changes for any of the special ipython syntaxes of `!`, `%`, `%%`, `?` etc. I guess this is acceptable as long as it's docuemented, however?

**Comment by ggrrll** [comment]
>any comment would become a markdown cell? 

yeah, that's my idea. - but I don't have long experience... so, other researchers might prefer other solutions

>Something like `to new notebook`?

that sounds very  cool :)...

(for the rest I can't help sorry...I have no experience)
